pandas/core/series.py

```diff
@@ -1522,40 +1555,157 @@ def unique(self):
 
         Examples
         --------
-        >>> pd.Series([2, 1, 3, 3], name='A').unique()
-        array([2, 1, 3])
+        If `raw=False`, the output is a Series:
+
+        >>> pd.Series([1, 1, 3, 2], name='A').unique(raw=False)
+        0    1
+        2    3
+        3    2
+        Name: A, dtype: int64
+
+        If `raw=True`, the output is an ndarray (if the data is not
+        categorical):
+
+        >>> pd.Series([1, 1, 3, 2], name='A').unique(raw=True)
+        array([1, 3, 2])
+
+        This method also deals well with timestamps,
 
-        >>> pd.Series([pd.Timestamp('2016-01-01') for _ in range(3)]).unique()
-        array(['2016-01-01T00:00:00.000000000'], dtype='datetime64[ns]')
+        >>> pd.Series([pd.Timestamp('2016-01-01')
+        ...            for _ in range(3)]).unique(raw=False)
+        0   2016-01-01
+        dtype: datetime64[ns]
+
+        as well as timezones:
 
         >>> pd.Series([pd.Timestamp('2016-01-01', tz='US/Eastern')
-        ...            for _ in range(3)]).unique()
-        array([Timestamp('2016-01-01 00:00:00-0500', tz='US/Eastern')],
-              dtype=object)
+        ...            for _ in range(3)]).unique(raw=False)
+        0   2016-01-01 00:00:00-05:00
+        dtype: datetime64[ns, US/Eastern]
 
         An unordered Categorical will return categories in the order of
         appearance.
 
-        >>> pd.Series(pd.Categorical(list('baabc'))).unique()
+        >>> pd.Series(pd.Categorical(list('baabc'))).unique(raw=False)
+        0    b
+        1    a
+        4    c
+        dtype: category
+        Categories (3, object): [b, a, c]
+
+        >>> pd.Series(pd.Categorical(list('baabc'))).unique(raw=True)
         [b, a, c]
         Categories (3, object): [b, a, c]
 
         An ordered Categorical preserves the category ordering.
 
         >>> pd.Series(pd.Categorical(list('baabc'), categories=list('abc'),
-        ...                          ordered=True)).unique()
-        [b, a, c]
+        ...                          ordered=True)).unique(raw=False)
+        0    b
+        1    a
+        4    c
+        dtype: category
         Categories (3, object): [a < b < c]
-        """
-        result = super(Series, self).unique()
```

Comment:
The branch for `raw=True` is exactly the same as before, but the diff is messed up because of the changed indentation.